versioned/storage/store/src/main/java/org/projectnessie/versioned/storage/versionstore/VersionStoreImpl.java

```diff
@@ -491,15 +502,15 @@ public PaginationIterator<KeyEntry> getKeys(Ref ref, String pagingToken, boolean
     IndexesLogic indexesLogic = indexesLogic(persist);
     StoreIndex<CommitOp> index = indexesLogic.buildCompleteIndex(head, Optional.empty());
 
-    StoreKey begin =
-        pagingToken != null && !pagingToken.isEmpty()
-            ? keyFromString(fromString(pagingToken).token().toStringUtf8())
-            : null;
-    StoreKey end = null;
-
-    Iterator<StoreIndexElement<CommitOp>> result = index.iterator(begin, end, false);
+    Iterator<StoreIndexElement<CommitOp>> result =
+        index.iterator(keyRanges.beginStoreKey(), keyRanges.endStoreKey(), false);
```

Comment:
If we do not derive an end key from `prefix`, then prefix queries will have to scan all the way to the end of the index, will they not?

Reply:
Fixed